frontend/awx/administration/notifiers/hooks/useNotificationsWatch.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react';
+import { RunningNotificationsType } from './useNotifiersRowActions';
+import { NotificationTemplate } from '../../../interfaces/NotificationTemplate';
+import { usePageAlertToaster } from '../../../../../framework';
+import { StatusLabel } from '../../../../common/Status';
+import { useTranslation } from 'react-i18next';
+
+export function useNotificationsWatch(type: 'detail' | 'list') {
+  // key:value = notification_template_id:notification_id
+  const [runningNotifications, setRunningNotifications] = useState<RunningNotificationsType>({});
+  const alertToaster = usePageAlertToaster();
+  const { t } = useTranslation();
+
+  return {
+    runningNotifications,
+    setRunningNotifications,
+    onNotifierStartTest: (template_id: string, notificationId: string) => {
+      const obj = { ...runningNotifications };
+      obj[template_id] = notificationId;
+      setRunningNotifications(obj);
+    },
+    checkNotifiers: (notificationTemplate: NotificationTemplate) => {
+      // search for notification id
+      const notificationId = runningNotifications[notificationTemplate.id.toString()];
+      const notifications = notificationTemplate.summary_fields?.recent_notifications;
+      const notification = notifications.find(
+        (notification) => notification.id.toString() === notificationId
+      );
+
+      if (notification && notification.status !== 'pending') {
+        if (type === 'detail') {
```

Comment:
Did you mean to show toast for list view like this?
```suggestion
        if (type === 'list') {
```
In the old UI the toast is shown on the list view and not on the details view

Reply:
Also saw the PR description, but feel like it's possible to have multiple notifications pop up for different notifications. Here's what I got when running two notifs after implementing the change suggested above
<img width="1661" alt="Screenshot 2024-04-26 at 11 49 00 AM" src="https://github.com/ansible/ansible-ui/assets/112966709/6f2c84ec-9ac1-487e-bd14-6b3c33132160">

Reply:
Showing it in both is fine too in my opinion